src/LogikUI/File/FileManager.cs

```diff
@@ -0,0 +1,337 @@
+                {
+
+                    xmlWriter.WriteStartElement("wire");
+
+                    //from
+                    xmlWriter.WriteStartElement("from");
+                    xmlWriter.WriteAttributeString("x", wire.Pos.X.ToString());
+                    xmlWriter.WriteAttributeString("y", wire.Pos.Y.ToString());
+                    xmlWriter.WriteEndElement();
+
+                    //to
+                    xmlWriter.WriteStartElement("to");
+                    xmlWriter.WriteAttributeString("x", wire.EndPos.X.ToString());
+                    xmlWriter.WriteAttributeString("y", wire.EndPos.Y.ToString());
+                    xmlWriter.WriteEndElement();
+
+                    xmlWriter.WriteEndElement();
+
+                }
+               
+                xmlWriter.WriteEndElement();
+
+                #endregion
+
+                #region Parse Components
+
+                xmlWriter.WriteStartElement("components");
+
+                foreach (var component in Components)
+                {
+                    xmlWriter.WriteStartElement("component");
+                    
+                    xmlWriter.WriteEndElement();
+                }
+
+                xmlWriter.WriteEndElement();
+
+                #endregion
+
+                #region Parse Labels
+
+                xmlWriter.WriteStartElement("labels");
+
+                foreach (var text_label in Labels)
+                {
+                    xmlWriter.WriteStartElement("label");
+                    xmlWriter.WriteAttributeString("size", text_label.TextSize.ToString());
+
+                    //text
+                    xmlWriter.WriteStartElement("text");
+                    xmlWriter.WriteString(text_label.Text);
+                    xmlWriter.WriteEndElement();
+
+                    xmlWriter.WriteEndElement();
+                }
+                
+                xmlWriter.WriteEndElement();
+
+                #endregion
+
+                xmlWriter.WriteEndElement();
+                xmlWriter.WriteEndDocument();
+                xmlWriter.Close();
+            }
+            catch (Exception e)
+            {
+                if (e is XmlException || e is InvalidOperationException)
+                {
+                    MessageDialog md = new MessageDialog(null, DialogFlags.DestroyWithParent, MessageType.Info, ButtonsType.Ok, "Unable to save project due to XML error.");
+                    md.Run();
+                    md.Dispose();
+
+                    throw new ArgumentException("Failed to parse to XML.", e);
+                } 
+                else
+                {
+                    MessageDialog md = new MessageDialog(null, DialogFlags.DestroyWithParent, MessageType.Info, ButtonsType.Ok, "Unable to save project due to access/write error.");
+                    md.Run();
+                    md.Dispose();
+
+                    throw new ArgumentException($"Failed to save to { filename }.", e);
+                }
```

Comment:
Once again there is no guarantee that this is because of a access problem.
We should give more explicit error messages in general.